spark/src/main/java/org/datacleaner/spark/SparkJobLauncher.java

```diff
@@ -60,18 +66,40 @@ public void launchJob(AnalysisJob analysisJob) {
         } else {
             JavaSparkContext sc = null;
             try {
-                DataContext dataContext = datastore.openConnection().getDataContext(); 
+                DataContext dataContext = datastore.openConnection().getDataContext();
                 if (dataContext instanceof CsvDataContext) {
                     CsvDataContext csvDataContext = (CsvDataContext) dataContext;
-                    String datastoreFilePath = csvDataContext.getResource().getQualifiedPath();
-            
+                    final Resource resource = csvDataContext.getResource();
+                    String datastoreFilePath = resource.getQualifiedPath();
+
                     SparkConf conf = new SparkConf().setAppName("DataCleaner-spark");
                     sc = new JavaSparkContext(conf);
 
-                    JavaRDD<String> instructionsRDD = sc.textFile(datastoreFilePath);
-                    long lineCount = instructionsRDD.count();
-        
-                    logger.info("Line count: " + lineCount);
+                    Accumulator<Integer> dataCleanerConfigurationReadsCounter = sc.accumulator(0,
+                            "DataCleanerConfigurationConfiguration reads counter");
+                    Accumulator<Integer> analysisJobReadsCounter = sc.accumulator(0, "AnalysisJob reads counter");
+
+                    SparkDataCleanerContext sparkDataCleanerContext = new SparkDataCleanerContext(_dataCleanerConfigurationPath, analysisJobXmlPath);
+                    sparkDataCleanerContext.addAccumulator(dataCleanerConfigurationReadsCounter);
+                    sparkDataCleanerContext.addAccumulator(analysisJobReadsCounter);
```

Comment:
Would it not be nicer to just add SparkContext to SparkDataCleanerContext? Then you don't need to create the accumulators up front here, but just create them when/if requested. And then you can also support more than those two specific accumulators.